### DISCUSSION TITLE
Best way to pass API key to plugin?

### DISCUSSION THREAD

**Discussion by Swimburger** [microsoft/semantic-kernel#2755] [Q&A]
I'm developing a plugin that relies on a 3rd party service that needs an API key. 
Currently, I pass the API key as a constructor parameter and store it in a field.
I also create and dispose an HTTP client as needed but would rather rely on the DI systems that .NET usually has.
I'm just not sure what the preferred approach is when developing SK plugins.

**Comment by dawwa** [comment]
Hi @teresaqhoang, would you please help to point the sample code in chat-copilot around auth handler callback? thanks.

**Comment by markwallace-microsoft** [comment]
I am cleaning up old unanswered discussions
